Plugins/ASEP/Get-ASEP.ps1

```diff
@@ -0,0 +1,176 @@
+
+    #Autorunsc executable locations
+    $Autorunsc64 = "$PRRoot\Bin\autorunsc64.exe"
+    $Autorunsc32 = "$PRRoot\Bin\autorunsc.exe"
+
+    #Verify binaries exist in Bin
+    $64bitTestPath = Get-Item -Path $Autorunsc64 -ErrorAction SilentlyContinue
+    $32bitTestPath = Get-Item -Path $Autorunsc32 -ErrorAction SilentlyContinue
+
+    if (-not $64bitTestPath) {
+
+        Write-Error "Autorunsc64.exe not detected in Bin. Place 64bit executable in Bin directory and try again."
+        Exit 
+
+    } elseif (-not $32bitTestPath) {
+
+        Write-Error "Autorunsc.exe not detected in Bin. Place 32bit executable in Bin directory and try again."
+        Exit
+
+    }
+
+    #Loop through machines in $ComputerName to obtain data for each machine (if multiple machines are specified)
+    foreach ($Computer in $ComputerName) {
+
+    #Handle $ComputerName defined as localhost and not break the Plugin
+    if ($Computer -match "Localhost") {
+
+        $Computer = $ENV:ComputerName
+    }
+    
+    #Verify machine is online and ready for data collection
+    if (-not (Test-Connection -ComputerName $Computer -Count 1 -Quiet)) {
+        
+        Write-Error ("{0} appears to be offline. Cannot gather ASEP data." -f $Computer)
+        Continue
+    }
+
+    #Determine system architecture and select proper Autorunsc executable
+    try {
+        $Architecture = (Get-WmiObject -ComputerName $Computer -Class Win32_OperatingSystem -Property OSArchitecture -ErrorAction SilentlyContinue).OSArchitecture
+        
+        if ($Architecture -eq "64-bit") {
+
+            $Installexe = $Autorunsc64
+
+        } elseif ($Architecture -eq "32-bit") {
+
+            $Installexe = $Autorunsc32
+
+        } else {
+            
+            Write-Error ("Unknown system architecture ({0}) detected for {1}. Data was not gathered.)" -f $Architecture, $Computer)
+            Continue
+        }
+    } catch {
+        
+        Write-Error ("Unable to determine system architecture for {0}. Data was not gathered." -f $Computer)
+        Continue
+    }
+
+    #Copy Autorunsc to remote host
+    $SmbPath = ("\\{0}\c`$\ProgramData\{1}" -f $Computer, (Split-Path -Path $Installexe -Leaf))
+    try {
+        
+        Copy-Item -Path $Installexe -Destination $SmbPath -ErrorAction SilentlyContinue
+        $RemoteFile = Get-Item -Path $SmbPath -ErrorAction SilentlyContinue
+
+        # verify that the file copy succeeded to the remote host
+        if (-not $RemoteFile) {
+            
+            Write-Error ("Remote file not found on {0}. There may have been a problem during the copy process. Data was not gathered." -f $Computer)
+            Continue
+        }
+    } catch {
+        
+        Write-Error ("An unexpected error occurred while copying Autorunsc to {0}. Data was not gathered" -f $Computer)
+        Continue
+    }
+
+
+    #Run Autorunsc on the remote host and collect ASEP data
+    
+    if ($Architecture -eq "64-bit") {
```

Comment:
You could simplify this section by using the `$Installexe` variable you set up earlier:

Could look like: `Invoke-Command -ComputerName $Computer -ScriptBlock { & C:\ProgramData\$Installexe /accepteula -a * -h -nobanner -v -s -t -c }`

Reply:
I initially tried to use this variable and the script block section does not appear to be able to expand that out. you get an error regarding the type of object you are trying to pass to the script block of invoke-command

Reply:
Huh weird I would expect you could expand things with that.

Looks like either of these might work:
`Invoke-Command -ComputerName $Computer -ScriptBlock { param($exe) & C:\ProgramData\$exe /accepteula -a * -h -nobanner -v -s -t -c } -ArgumentList $Installexe`

or
```
$ScriptBlock = $ExecutionContext.InvokeCommand.NewScriptBlock("& C:\ProgramData\$Installexe /accepteula -a * -h -nobanner -v -s -t -c")
Invoke-Command -ComputerName $Computer -ScriptBlock $ScriptBlock
```

Reply:
Done